### DISCUSSION TITLE
Working hours

### DISCUSSION THREAD

**Discussion by michalzubkowicz** [jordwest/news-feed-eradicator#385] [Ideas]
Something like "working hours" would be usable. I can be distracted and procrastinated on weekends or evenings.

**Comment by marksnell** [comment]
Even just an option to turn it off when you do want to see the newsfeed.

**Comment by michalzubkowicz** [comment]
It make no sense when you can disable it quickly. It's sometimes very hard to not launch another browser. I've got rule that I can check FB only on phone, and I'm placing phone away from my hand when I'm working :)

**Comment by jordwest** [comment]
Doing some experimentation in #31

**Comment by wdaher** [comment]
I'm definitely interested in the toggle timer functionality as per #31 . (There used to be a facebook news feed blocking extension that I liked that supported it, though it seems to have disappeared.)

I suspect this functionality would also scratch @jntrcs 's  Request #39

**Comment by pburtchaell** [comment]
I'm really interested in this feature! The plugin is great to use for the majority of the time, but there's also times when I genuinely want to use the news feed to keep up with friends and family. 

There's two ways I could see this working:
- I see news feed up to a certain limit (e.g., rules like "10 minutes per day")
- I see news feed in certain time frames (e.g., "working hours")

@jordwest Do you need help with #31? I can give this some time if you need it.

**Comment by Bejasc** [comment]
@jordwest @pburtchaell Did you guys make any progress on this? I notice #31 is closed without comment.

**Comment by maxsu** [comment]
I suspect that working hours is a bit complicated, since it requires a whole scheduling systems. As an easier to implement yet still viable alternative I've proposed #74 : Simple Extension toggle. I'd like to get that implemented and see where it goes from there. Feel free to join the discussion - especially looking for folks who can help me plan the code change and feature scope (@michalzubkowicz, @jordwest, @pburtchaell, @wdaher  - would be honored by any suggestions you can add)

**Comment by michalzubkowicz** [comment]
@maxsu 

I'm thinking that the purpose of this plugin is to not procrastinate :), if you make a button which will easily override everything, many persons will have the urge to press it. 

I don't think that the whole scheduling system is needed (something like cron). You can easily create a table with hours which can be selected or not on a given day and then check the hour with simple switch in code.
